### PR TITLE
ADLS file system upgrade

### PR DESCRIPTION
**Pull request by xiangfu0**
1. upgrade azure-data-lake-store-sdk version to 2.3.10
2. use DataLakeFileSystemClient for open method as well
3. support a new auth type `NONE`, meaning no cred is added.

### CODE REVIEW

File: pinot-plugins/pinot-file-system/pinot-adls/src/main/java/org/apache/pinot/plugin/filesystem/ADLSGen2PinotFS.java [hunk 1]
```diff
@@ -191,20 +182,22 @@ public void init(PinotConfiguration config) {
         clientSecretCredentialBuilder.httpClient(builder.build());
 
         dataLakeServiceClientBuilder.credential(clientSecretCredentialBuilder.build());
-        blobServiceClientBuilder.credential(clientSecretCredentialBuilder.build());
+        break;
+      }
+      case NONE: {
```
Comment (snleee) [inline pinot-plugins/pinot-file-system/pinot-adls/src/main/java/org/apache/pinot/plugin/filesystem/ADLSGen2PinotFS.java:187]: `ANONYMOUS_ACCESS` may be a better naming for authentication type?

File: pinot-plugins/pinot-file-system/pinot-adls/src/main/java/org/apache/pinot/plugin/filesystem/ADLSGen2PinotFS.java [hunk 2]
```diff
@@ -191,20 +182,22 @@ public void init(PinotConfiguration config) {
...
+        LOGGER.info("Authenticating using anonymous access");
         break;
       }
       default:
-        throw new IllegalStateException("Expecting valid authType. One of (ACCESS_KEY, AZURE_AD, AZURE_AD_WITH_PROXY");
+        throw new IllegalArgumentException(
```
Comment (snleee) [inline pinot-plugins/pinot-file-system/pinot-adls/src/main/java/org/apache/pinot/plugin/filesystem/ADLSGen2PinotFS.java:192]: Should we try to initialize the client using `DefaultAzureCredential` to have a way to pass the credentials to azure client using ENV variables (or system properties)?

File: pinot-plugins/pinot-file-system/pinot-adls/src/main/java/org/apache/pinot/plugin/filesystem/ADLSGen2PinotFS.java [hunk 3]
```diff
@@ -586,15 +579,8 @@ public boolean touch(URI uri)
...
-    // Use Blob API since read() function from Data Lake Client currently takes "OutputStream" as an input and
-    // flush bytes to an output stream. This needs to be piped back into input stream to implement this function.
-    // On the other hand, Blob API directly allow you to open the input stream.
-    BlobClient blobClient = _blobServiceClient.getBlobContainerClient(_fileSystemClient.getFileSystemName())
-        .getBlobClient(AzurePinotFSUtil.convertUriToUrlEncodedAzureStylePath(uri));
-
-    return blobClient.openInputStream();
-    // Another approach is to download the file to the local disk to a temp path and return the file input stream. In
-    // this case, we need to override "close()" and delete temp file.
+    return _fileSystemClient.getFileClient(AzurePinotFSUtil.convertUriToUrlEncodedAzureStylePath(uri)).openInputStream()
```
Comment (snleee) [inline pinot-plugins/pinot-file-system/pinot-adls/src/main/java/org/apache/pinot/plugin/filesystem/ADLSGen2PinotFS.java:582]: Thanks for addressing this!

**Review by snleee — APPROVED** [review]
LGTM otherwise

File: pinot-plugins/pinot-file-system/pinot-adls/src/main/java/org/apache/pinot/plugin/filesystem/ADLSGen2PinotFS.java (hunk 1 above)
Comment (xiangfu0) [reply pinot-plugins/pinot-file-system/pinot-adls/src/main/java/org/apache/pinot/plugin/filesystem/ADLSGen2PinotFS.java]: done

File: pinot-plugins/pinot-file-system/pinot-adls/src/main/java/org/apache/pinot/plugin/filesystem/ADLSGen2PinotFS.java (hunk 2 above)
Comment (xiangfu0) [reply pinot-plugins/pinot-file-system/pinot-adls/src/main/java/org/apache/pinot/plugin/filesystem/ADLSGen2PinotFS.java]: Make sense, updated the default behavior to use `DefaultAzureCredentialBuilder`: https://learn.microsoft.com/en-us/java/api/com.azure.identity.defaultazurecredentialbuilder?view=azure-java-stable

**Comment by xiangfu0** [comment]
@jackjlli please also check, there is a behavior change for handling the default scenario for the AzurePinotFS. Since it's on the failover code path, I won't mark it as backward incompatible.

**Pull request merged**